### PR TITLE
Error code changed BAD_REQUEST to FORBIDDEN

### LINKED ISSUE #636: Fix: refactor HTTP error codes GET /mentorship_relation/{request_id}/task/{task_id}/comments
## Description
As a GSoC student,
I need the correct HTTP error response code to be returned accordingly by Mentorship System API in relation to GET /mentorship_relation/{request_id}/task/{task_id}/comments, 
so that I can represent the error message accordingly to both Mentorship System and Bridge In Tech users.

## Mocks
NA

### Update [Required]
- [ ] 403: Forbidden needs to be returned instead of 400: Bad Request if it has unaccepted state relation

### Enhancement to Update [Optional]
NA
## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
1 hour

### PR DESCRIPTION
**Pull request by b-thebest**
…tion_id/task/<task_id>/comments

### Description

Fixes #636

### Type of Change:

- Code: error code changed from BAD_REQUEST to FORBIDDEN

**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)

### How Has This Been Tested?
Tested locally 

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas

**Code/Quality Assurance Only**

- [x] My changes generate no new warnings 
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by devkapilbansal — DISMISSED** [review]
LGTM :+1:

**Comment by devkapilbansal** [comment]
@b-thebest please update your PR title and commit message by following the Commit message guidelines

**Comment by b-thebest** [comment]
@devkapilbansal I have done some changes, These changes are correct? Please correct me if I am wrong

**Review by devkapilbansal — APPROVED** [review]
LGTM :+1: 
Thanks @b-thebest  for this

**Review by vj-codes — DISMISSED** [review]
@b-thebest thank you for your contribution:)
LGTM 🎉

**Comment by devkapilbansal** [comment]
My bad, @b-thebest this PR needs changes

**Review by devkapilbansal — CHANGES_REQUESTED** [review]
Change this to Forbidden too
https://github.com/anitab-org/mentorship-backend/blob/fffb4c39027b042f392d4edf0bb462e522cb7320/app/api/dao/task_comment.py#L21

**Comment by b-thebest** [comment]
@devkapilbansal It's done, but some checks are getting failed. Is it fine ?

**Review by devkapilbansal — DISMISSED** [review]
LGTM 👍 
@b-thebest codecov/patch failed because these lines are not covered. #989 in already opened for that

**Comment by devkapilbansal** [comment]
@vj-codes please review it again

**Review by epicadk — DISMISSED** [review]
💯

**Comment by devkapilbansal** [comment]
@b-thebest one last request.
Please rebase your branch with current develop. Then you may need to make changes in some tests too.

**Comment by devkapilbansal** [comment]
Once that is done, I will add label `Ready to Merge` since I tested too and automated tests will also ensure its working

**Comment by b-thebest** [comment]
@devkapilbansal Sorry for being late. I will do it and let you know soon

**Comment by b-thebest** [comment]
> @b-thebest one last request.
> Please rebase your branch with current develop. Then you may need to make changes in some tests too.

Could you please help me in that, sometimes I get stuck in rebasing.

**Comment by vj-codes** [comment]
@b-thebest any updates here?

**Comment by b-thebest** [comment]
I have made pull request. Could you please help me in rebasing my
repository with remote ?

**Comment by vj-codes** [comment]
@b-thebest `git checkout develop` and then `git merge branchname `should work ?
cc : @epicadk

**Comment by b-thebest** [comment]
My changes in new branch and pull request will still be there after
rebasing?

**Comment by devkapilbansal** [comment]
@b-thebest to rebase your branch just follow:-
```
git fetch upstream develop
git rebase upstream/develop
git push -f
```

Your PR and changes done will remain after rebasing

**Comment by vj-codes** [comment]
@b-thebest any updates? you can also do this via github

**Comment by b-thebest** [comment]
Sorry for being late. My health conditions were not good during past days. I will be completing it very soon and share update as soon as possible.

**Comment by b-thebest** [comment]
Please review this PR and close the issue

**Review by epicadk — DISMISSED** [review]

**Review by vj-codes — DISMISSED** [review]
Thank you for your contribution @b-thebest 🎉

File: app/api/resources/task.py [hunk 1]
```diff
@@ -23,7 +23,7 @@ class CreateTask(Resource):
     @task_ns.doc("create_task_in_mentorship_relation")
     @task_ns.expect(auth_header_parser, create_task_request_body)
     @task_ns.response(HTTPStatus.CREATED, f"{messages.TASK_WAS_CREATED_SUCCESSFULLY}")
-    @task_ns.response(HTTPStatus.FORBIDDEN, f"{messages.UNACCEPTED_STATE_RELATION}")
+    @task_ns.response(HTTPStatus.BAD_REQUEST, f"{messages.UNACCEPTED_STATE_RELATION}")
```
Comment (devkapilbansal) [inline app/api/resources/task.py]: Why this change is made??

Comment (b-thebest) [reply app/api/resources/task.py]: I think unaccepted relation must fall in BAD_REQUEST. When unauthorized user try to do something then FORBIDDEN is required

Comment (epicadk) [reply app/api/resources/task.py]: Nope should be forbidden. Nice catch @devkapilbansal . Bad request would imply that the server cannot understand the request, however here the server understands the request but refuses to process it.

Comment (b-thebest) [reply app/api/resources/task.py]: I changed it again to FORBIDDEN. Please check
@devkapilbansal @epicadk

**Review by devkapilbansal — DISMISSED** [review]
LGTM :+1: 

Thanks @b-thebest for your contribution

**Review by isabelcosta — DISMISSED** [review]
Great work @b-thebest ! Thank you for contributing!

**Comment by isabelcosta** [comment]
The changes made in this PR were tested locally. Following are the results:

1. Code review - Done

2. All possible responses (positive and negative tests) were tested as below:

   * _Get comments from a task of a non accepted relation_  
     _Screenshot/gif_:  
image

     _Expected Result_:  403 is returned when fetching comments from a not accepted relation tasks
     _Actual Result_: 403 is returned as expected

  
4. Additional Comments: https://ms-backend-review-pr-1030.herokuapp.com/

5. Status of PR Changed to: Ready to Merge.

**Comment by isabelcosta** [comment]
will merge, because the approvals were revoked by my merge conflict resolve
